cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -188,9 +191,20 @@ func (na *nodeAnnotator) processNextWorkItem() bool {
 	}
 	defer na.queue.Done(key)
 
-	na.sync(key.(string))
+	err := na.sync(key.(string))
+	if err != nil {
+		// Drop from the queue if failing too many times.
+		if na.queue.NumRequeues(key) > nodeAnnotatorQueueRetryLimit {
+			klog.Errorf("Sync %v exceeded retry limit, last err: %v", key, err)
+			na.queue.Forget(key)
+			return true
+		}
+		klog.Warningf("Requeue %v (%v times) due to err: %v", key, na.queue.NumRequeues(key), err)
+		na.add(key)
```

Comment:
This function expects a nod object, not a key. Stupid go with no generics doesn't allow for type safety here :(

https://github.com/kubernetes/cloud-provider-gcp/blob/625d1222bf8305ba01c9cdccb41fbd71ffbde762/cmd/gcp-controller-manager/node_annotator.go#L168

Reply:
Call AddRateLimited directly.

Reply:
Yep just noticed the the type was wrong.